build-system/server.js

```diff
@@ -33,6 +33,14 @@ var url = require('url');
 app.use(bodyParser.json());
 app.use(morgan('dev'));
 
+app.use('/', function(req, res, next) {
+  res.setHeader('AMP-Access-Control-Allow-Source-Origin',
```

Comment:
By reading [AMP CORS](https://github.com/ampproject/amphtml/blob/master/spec/amp-cors-requests.md#cors-security-in-amp), seems if local server wants to trust all XHRs, it can read the source origin from URL param `__amp_source_origin`, and set it to the response header.

Did you try if that works? In this way, only CORS XHR will have that header, and also there's no need to hard code "localhost".

I'm not sure if we have tests or examples for CORS auth failure case. In that case, we need to explicitly blacklist those resources to remove the header. You can run all the integration test and check our manual examples that use CORS XHR to see if your changes will break any.